### PR TITLE
Set SendMessageRequest.Config.ReturnImmediately from AllowBackgroundResponses on the A2A client

### PR DESCRIPTION
**Pull request by PratikDhanave**
## What

The A2A client provider (`provider/a2aprovider/a2a.go`) built the `a2a.SendMessageRequest` with no `Config`, so `run()` never read the `AllowBackgroundResponses` option. This change reads the option and sets `SendMessageConfig.ReturnImmediately` on the non-streaming send.

## Why

`AllowBackgroundResponses` is a real client option that is validated generically in `agent.go` (it requires a session). A caller passing `agent.AllowBackgroundResponses(true)` therefore passed validation, but the A2A provider silently ignored it, so the request never asked the remote agent to return immediately after creating the task. `openaiprovider/responses.go` already reads the same option correctly.

This mirrors .NET `A2AAgent.RunCoreAsync`, which sets the send `Configuration` only on the non-streaming send; the streaming `SendStreamingMessage` path is left unchanged.

## Tests

Added `TestRunAllowBackgroundResponsesSetsReturnImmediately` to `a2a_test.go`, reusing the existing `mockA2ATransport` that captures `capturedMessageSendParams`:
- with `AllowBackgroundResponses(true)` + a session, asserts `Config.ReturnImmediately == true`;
- without the option, asserts `Config` is nil or `ReturnImmediately == false`.

The test fails before the fix and passes after. `go build ./...`, `go vet ./provider/a2aprovider/...`, and `go test ./provider/a2aprovider/...` all pass.

### CODE REVIEW

File: provider/a2aprovider/a2a.go [hunk 1]
```diff
@@ -101,6 +101,10 @@ func (a *a2aProvider) run(ctx context.Context, messages []*message.Message, opti
 			if stream {
 				seq = a.client.SendStreamingMessage(ctx, params)
 			} else {
+				// Mirror .NET A2AAgent.RunCoreAsync, which sets the send
+				// configuration only on the non-streaming send.
+				allowBackground, _ := agent.GetOption(options, agent.AllowBackgroundResponses)
+				params.Config = &a2a.SendMessageConfig{ReturnImmediately: allowBackground}
```
[the remark this answers is not in the document]
Comment (PratikDhanave) [reply provider/a2aprovider/a2a.go]: Good catch — fixed in 4db7c47: Config is now only set when AllowBackgroundResponses is explicitly enabled (with ReturnImmediately: true), leaving Config nil in the default case so the wire request is unchanged. Also tightened the disabled-case test to assert Config is nil.

**Review by gdams — APPROVED** [review]
